sumpy/assignment_collection.py

```diff
@@ -191,6 +191,10 @@ def run_global_cse(self, extra_exprs=[]):
             assert isinstance(name, sym.Symbol)
             self.add_assignment(name.name, value)
 
+        for name, new_expr in zip(assign_names, new_assign_exprs):
+            del self.assignments[name]
+            self.assignments[name] = new_expr
```

Comment:
IIUC, this is semantically a no-op. Add a comment on why it helps?

Reply:
Done in ca4cedd09e50b208a4b636f59de947ccabfa9d41